### ISSUE TITLE
Panel Layout isn't remembered

### ISSUE THREAD

**Issue by Plagiatus** [JirkaDellOro/FUDGE#403]
## The issue

Neither the layout between panels nor the layout of the views within the panel.

-------

## Some relevant info

> Configs are now strongly typed. In addition, GoldenLayout now has "Configs" and "Resolved Configs"
> - Configs
>   Application developers will mainly work with "Configs". A "Config" supports optional properties. If a property is not specified, a default will be used. In addition, "Config" also will handle backwards compatibility. It will migrate deprecated properties to their new values.
    Config parameters in GoldenLayout API methods will be of type "Config". The one exception is LayoutConfig.saveLayout() which returns a "Resolved Config".
> - Resolved Configs
    Golden-Layout internally uses "Resolved Config"s. Whenever an API function is passed a "Config", GoldenLayout will resolve it to its corresponding "Resolved Config". This resolving process will set default values where an optional value has not been specified. It will also handle backwards compatibility. This allows the GoldenLayout library to always work with fully configured Configs.
> 
> For persistence of configs, always save the "Resolved Config" returned by LayoutManager.saveLayout(). When reloading a saved Layout, first convert the saved "Resolved Config" to a "Config" by calling LayoutConfig.fromResolved().

from: https://github.com/golden-layout/golden-layout/blob/master/docs/version-2/index.md

&nbsp;

To listen to when the layout changes to save the state you should be able to listen to and handle it like this:
```ts
Page.goldenLayout.on("stateChanged", Page.hndStateChanged);

private static hndStateChanged(){
  let state = JSON.stringify(Page.goldenLayout.rootItem.layoutManager.saveLayout());
}
```
However, that doesn't seem to fire all the time and also doing the reverse doesn't seem to have any effect.
```ts
Page.goldenLayoutModule.LayoutConfig.fromResolved(state);
```

&nbsp;

I also realized that the gl loading is a bit convoluted. `Page.loadLayout()` is called multiple times, which only sets a base config. Then on `Project.load()` the various panels are loaded/added in `Page.setPanelInfo()`.

&nbsp;

So, in conclusion: To add this functionality, we probably need to figure out how to save the panel layout into the `Panel` and the overall layout of the project and somehow restore it. However, since it took me way too long to figure out these things and combining through the GL documentation is tedious, I will be tackling other issues for now.
